app/main.py

```diff
@@ -1,3 +1,63 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: int) -> Distance:
+        if isinstance(other, int | float):
+            return Distance(km=self.km * other)
+
+    def __truediv__(self, other: int) -> Distance:
+        if isinstance(other, (int, float)) and other != 0:
+            return Distance(round(self.km / other, 2))
```

Comment:
The `__truediv__` method should ensure that the result is an integer since `km` is expected to be an integer. Consider using `int(self.km / other)` instead of `round(self.km / other, 2)`.